### PR TITLE
chore(deps): Upgrade Rust to 1.61.0

### PR DESCRIPTION
**Pull request by lukesteensen**

### CODE REVIEW

**Review by bruceg — APPROVED** [review]

File: lib/value/src/value.rs [hunk 1]
```diff
@@ -100,7 +100,7 @@ impl PartialEq<Self> for Value {
             }
             (Value::Integer(a), Value::Integer(b)) => a.eq(b),
             (Value::Object(a), Value::Object(b)) => a.eq(b),
-            (Value::Null, Value::Null) => true,
+            (Self::Null, Self::Null) => true,
```
Comment (JeanMertz) [inline lib/value/src/value.rs:103]: This seems.... odd. Does it only complain about this specific one? If that's the case, I'd rather disable this Clippy lint. I'm fine using `Self` consistently, but it's just confusing to mix the two styles.

**Comment by lukesteensen** [comment]
@spencergilbert I would say progress is stalled at the moment. We still have similar issues with cross compilation that we had with the failed 1.59 upgrade, and I wasn't able to work out any obvious solutions.

**Comment by spencergilbert** [comment]
> @spencergilbert I would say progress is stalled at the moment. We still have similar issues with cross compilation that we had with the failed 1.59 upgrade, and I wasn't able to work out any obvious solutions.

👍 `kube-rs` upgrades are blocked by the current version, so I wanted to check in.

File: lib/value/src/value.rs (hunk 1 above)
Comment (spencergilbert) [reply lib/value/src/value.rs:103]: Any other opinions here? I can see both sides of the argument.

Comment (spencergilbert) [reply lib/value/src/value.rs:103]:
```
error: unnecessary structure name repetition
   --> lib/value/src/value.rs:103:14
    |
103 |             (Value::Null, Value::Null) => true,
    |              ^^^^^ help: use the applicable keyword: `Self`
    |
note: the lint level is defined here
   --> lib/value/src/lib.rs:7:5
    |
7   |     clippy::nursery,
    |     ^^^^^^^^^^^^^^^
    = note: `#[deny(clippy::use_self)]` implied by `#[deny(clippy::nursery)]`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#use_self

error: unnecessary structure name repetition
   --> lib/value/src/value.rs:103:27
    |
103 |             (Value::Null, Value::Null) => true,
    |                           ^^^^^ help: use the applicable keyword: `Self`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#use_self
```

Comment (jszwedko) [reply lib/value/src/value.rs:103]: I'd be 👍 on using `Self` everywhere here.

Comment (spencergilbert) [reply lib/value/src/value.rs:103]: Updated that and checking to see if it's angry about that at all

**Comment by spencergilbert** [comment]
@tobz spent a little time looking at why we're seeing missing event failures now - I didn't notice anything in the changelogs either. Mind taking a look when you can?

File: lib/value/src/value.rs (hunk 1 above)
Comment (lukesteensen) [reply lib/value/src/value.rs]: Yeah, I believe it was only catching the non-tuple, non-struct variant here but probably intends to cover all of them. I'd agree with updating to `Self` everywhere.

**Review by spencergilbert — APPROVED** [review]

**Comment by jszwedko** [comment]
🥳

**Comment by jszwedko** [comment]
Thanks to @tobz for tracking this one down!

**Pull request merged**